packages/lesswrong/lib/sql/UpdateQuery.ts

```diff
@@ -167,18 +167,31 @@ class UpdateQuery<T extends DbObject> extends Query<T> {
     value: any,
     format: (resolvedField: string, updateValue: Atom<T>[]) => Atom<T>[],
   ): Atom<T>[] {
-    try {
-      // If we're updating the value of a JSON blob without totally replacing
-      // it then we need to wrap the update in a call to `JSONB_SET`.
-      if (field.includes(".")) {
-        const updateValue = this.compileUpdateExpression(value, { skipTypeHint: true });
-        const {column, path} = this.buildJsonUpdatePath(field);
-        return format(
-          column,
-          ["JSONB_SET(", column, ",", path, "::TEXT[],", ...updateValue, ", TRUE)"],
-        );
+    // If we're updating the value of a JSON blob without totally replacing
+    // it then we need to wrap the update in a call to `JSONB_SET`.
+    if (field.includes(".")) {
+      const updateValue = this.compileUpdateExpression(value);
+      const {column, path} = this.buildJsonUpdatePath(field);
+      // Check if we're trying to unset the field (currently you can only unset fields in the first level)
+      if (value === null) {
+        const fieldTokens = field.split(".")
+        if (fieldTokens.length === 2) {
+          return format(
+            column,
+            [column, ` - '${fieldTokens[1]}'`],
+          );
+        } else {
+          throw new Error(`Unsetting a field past the first level of a JSON blob is not yet supported`)
```

Comment:
As far as I can tell, we don't unset past the first level anywhere so I decided to just fix the easy case